File: cpp/2302.count-subarrays-with-score-less-than-k.cpp
#include <vector>
using std::vector;

class Solution {
public:
  long long countSubarrays(vector<int> &nums, long long k) {
    long total = 0, cur = 0;

    for (int i = 0, j = 0; i < nums.size(); ++i) {
      cur += nums[i];

      while (j <= i && cur * (i - j + 1) >= k) {
        cur -= nums[j++];
      }

      total += i - j + 1;
    }

    return total;
  }
};

File: python/2302.count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: list[int], k: int) -> int:
        n = len(nums)
        i, j = 0, 0
        total, curSum = 0, 0

        for j in range(n):
            curSum += nums[j]

            while i <= j and curSum * (j - i + 1) >= k:
                curSum -= nums[i]
                i += 1

            total += j - i + 1

        return total

